osfs/os.go

```diff
@@ -23,6 +23,10 @@ func New(baseDir string) billy.Filesystem {
 	return chroot.New(&OS{}, baseDir)
 }
 
+type file struct {
```

Comment:
Maybe add a comment here about why this wrapper is needed?  Judging just based on this go file, it's pretty confusing.